Build_Process/GeneratingIso.md

```diff
@@ -0,0 +1,80 @@
+The generated iso can then be used as a cdrom in an emulator, or burnt to a real disk or usb thumb drive as you would for installing any other operating system.
+
+## Xorisso
+Xorisso is tool used to create iso disk images. Iso is actually quite a complex format, as it aims to be compatable with a lot of different formats and ways of booting.
+
+A walkthrough of xorisso is outside the scope of this section, but just know it's the standard tool for working with the iso format.
+
+## Grub (Multiboot 2)
+Grub provides a tool called `grub-mkrescue`. Originally it was intended for making rescue disks (hence the name), but it works just fine for our purposes.
+
+The tool is very straight forward, you create a root folder and populate that with what you want to appear on the final iso. In our case we'll need `grub.cfg` and our kernel (called `kernel.elf` for now), so we'll create the following directory layout:
+
+```
+disk/
+  \-boot/
+      |-grub/
+      |   \-grub.csfg
+      \-kernel.elf
+```
+
+Now we can run `grub-mkrescue -o my_iso.iso disk`, and it will generate an iso called `my_iso.iso` with the contents of `disk/` as the root directory.
+
+### Grub.cfg
+This file contains some global config options for grub (setting boot timeouts, etc...) as well as a list of boot options for this disk. In our case we only have the one option of our kernel. 
+
+Global options can be changed using the `set option=value` syntax, and a list is available from the grub documentation.
+
+For each boot option you want to provide, you'll need to create a menu entry within grub.cfg. A menu entry consists of a header, and then a body containing a series of grub commands to boot your operating system. This allows grub to provide a flexible interface for more complex setups. 
+
+However in our case, we just want to boot our kernel using the standard multiboot2 protocol. Fortunately grub has a built in command (`multiboot2`) for doing just that:
+
+```
+menuentry "My Operating System" {
+    multiboot2 /boot/kernel.elf
+    boot
+}
+```
+
+Note the last command `boot`, this tells grub to complete the boot process and actually run our kernel.
+
+## Limine (Stivale 2, multiboot 2)
+Limine requires us to build the iso itself, and then provides a tool to install itself onto the iso. 
+
+To get started we'll want to create a working directory to use as the root of our iso. In this case we'll use `disk/`. Next we'll need to clone the latest binary branch of limine (using `git clone https://github.com/limine-bootloader/limine.git --branch=v3.0-branch-binary --depth=1`) and copy 'limine.sys', 'limine-cd.bin', and 'limine-cd-efi.bin' into `disk/limine/`, creating that directory if it dosn't exist.
+
+Now we can copy our limine.cfg and kernel into your working directory. Limine.cfg needs to be in one of a few locations in order to be found, the best place is the root directory. 
+
+Now we're ready to run the following xorisso command:
+
+```sh
```

Comment:
this command is referred to limine case, why it is not added for grub too? 
(or btw since this is kind of common part, it can be placed into a pargraph of it's own otherwise someone using grub can miss it)

Reply:
no you are right. i think in the past i used xorriso to make an iso, but since you mentioned already grub-mkrescue i don´t think there is need for it